### ISSUE TITLE
Windows 7 + IE11 token issue

### ISSUE THREAD

**Issue by fkooman** [Amebis/eduVPN#42]
Clean, fully up to date Windows 7 with IE11 as default browser.

![screenshot from 2017-11-22 17-39-07](https://user-images.githubusercontent.com/616885/33139200-df8a94be-cfac-11e7-863b-ac8e811e0b35.png)
![screenshot from 2017-11-22 17-39-39](https://user-images.githubusercontent.com/616885/33139202-dfb7c146-cfac-11e7-8dcc-e646978a37f4.png)

```
eduOAuth.AccessTokenException: The provided authorization grant (e.g., authorization code, resource owner credentials) or refresh token is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.
"code" has invalid signature
   at eduOAuth.AccessToken.<FromAuthorizationServerResponseAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at eduOAuth.AuthorizationGrant.<ProcessResponseAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at eduVPN.ViewModels.AuthorizationPopup.<<get_Authorize>b__18_0>d.MoveNext()
```

Works fine if I install Firefox and make it the default browser.

**Comment by fkooman** [comment]
If I can venture a guess: it may be that the query parameter you get back, in this case `code` is malformed, possibly due to issues with URL encoding...

**Comment by rozmansi** [comment]
Wow, this happens with fully up-to-date Win7+IE11 too.

**Comment by rozmansi** [comment]
IE11 does not escape OAuth URI when it calls OAuth call-back.

IE: "org.eduvpn.app:/api/callback?code=EsxfyjdTlXx/1JJlINMQmkxyIETvp5JFM+2KzRRFyV+qv11qZ/AuRRUCnZEBDLPE5WiqJkJx0kaqJh5WnE..."

FF: "org.eduvpn.app:/api/callback?code=Ycb99ebuklDbpK9DPOU7b7e2g8jyWgwYG4wVbdxZxunoRt0lecmNmDW7iawZ4pKmQ5e9EnLBkw%2BnYFYv..."

The problem is "+" character when present in code parameter. Since the client unescapes the URI, "+" is converted to space. When the client re-escapes it to exchange the authorization code for the token, %20 is send instead of %2B.

**Comment by fkooman** [comment]
Attempting to fix this now in the OAuth server as to never give out authorization codes that require URL encoding. That's "cleaner" anyway.

**Comment by fkooman** [comment]
Fixed in `fkooman/oauth2-server` [2.0.1](https://github.com/fkooman/php-oauth2-server/blob/master/CHANGES.md#201-2017-11-29)
